src/tls_stream.rs

```diff
@@ -617,16 +617,12 @@ impl<S> TlsStream<S>
     }
 
     fn write_out(&mut self) -> io::Result<usize> {
-        let mut out = 0;
-        while self.out_buf.position() as usize != self.out_buf.get_ref().len() {
-            let position = self.out_buf.position() as usize;
-            let nwritten = try!(self.stream.write(&self.out_buf.get_ref()[position..]));
-            out += nwritten;
-            self.out_buf.set_position((position + nwritten) as u64);
-        }
-
-        Ok(out)
-    }
+        let mut nwritten = 0;
+		let position = self.out_buf.position() as usize;
+		nwritten += self.stream.write(&self.out_buf.get_ref()[position..])?;
```

Comment:
The general rule of thumb is to respect that rule as much as you can. We can't do that here since forcing a single write in this manner means the stream totally breaks if the kernel buffer happens to fill.